test/util/util.c

```diff
@@ -531,16 +531,32 @@ static bool _state_need_kms(_state_machine_t *state_machine, bson_error_t *error
 
         iov.iov_base = (char *)mongocrypt_binary_data(http_req);
         iov.iov_len = mongocrypt_binary_len(http_req);
-
         if (state_machine->trace) {
-            MONGOC_DEBUG("--> sending KMS message: \n%.*s", (int)iov.iov_len, (char *)iov.iov_base);
+            MONGOC_DEBUG("--> sending KMS message:");
+            const char *kms_provider = mongocrypt_kms_ctx_get_kms_provider(kms_ctx, NULL);
+            if (0 == strcmp(kms_provider, "kmip")) {
+                // Print KMIP protocol as hex.
+                uint8_t *as_u8_ptr = iov.iov_base;
+                for (size_t i = 0; i < iov.iov_len; i++) {
+                    printf("%02x", as_u8_ptr[i]);
```

Comment:
Should this `printf` (and below) be a `MONGOC_DEBUG`?

Reply:
No. `printf` was intended. `MONGOC_DEBUG` adds newlines and a timestamp. This print is intended to dump hex for easy copy-pasting.